### ISSUE TITLE
Jetty 10.0.14 uses wrong pathSpec for request

### ISSUE THREAD

**Issue by Or4ng3** [eclipse/jetty.project#9517]
**Jetty version(s)**
10.0.14

**Java version/vendor** `(use: java -version)`
17

**OS type/version**
Doesn't matter

**Description**

Related to #9055

We have two servlets:
1) "/*"
2) "/foo/*"

Prior to Jetty 10.0.14, requests to "/foobar/something" were handled by servlet 1. After the upgrade, requests are unable to reach it. 

After debugging I found that problem was related to org.eclipse.jetty.http.pathmap.PathMappings#getMatched method:

```
        // path = "/foobar/something"

        MappedResource<E> prefix = _prefixMap.getBest(path); // ServletPathSpec@1c0d1eaa{/foo/*}

        if (prefix != null)
        {
            MatchedPath matchedPath = prefix.getPathSpec().matched(path);     // null, since found prefix doesn't match path
            if (matchedPath != null)
                return new MatchedResource<>(prefix.getResource(), prefix.getPathSpec(), matchedPath);
        }
```

**Comment by joakime** [comment]
Oof!

This is a bad bug that somehow creeped in (don't we have unit test for this??)

**Comment by joakime** [comment]
I am able to replicate this with a simple test case.
Checking to see how far back this issue exists for.

**Comment by joakime** [comment]
Not able to replicate in Jetty 10.0.13 or older.

**Comment by joakime** [comment]
Opened PR #9518 to address this

**Comment by joakime** [comment]
We've merged this into `jetty-10.0.x` and `jetty-11.0.x` and will be cutting a new release shortly.
